src/main.rs

```diff
@@ -315,11 +315,7 @@ fn main() -> Result<i32> {
 
     if let Some(ref mut s) = session {
         let cookie_jar = cookie_jar.lock().unwrap();
-        // TODO: investigate why expires is set to None when get_request_cookies() is used
-        // but not for iter_any()
-        let response_cookies = cookie_jar.get_request_cookies(&url).collect::<Vec<_>>();
-        s.save_cookies(response_cookies);
-
+        s.save_cookies(cookie_jar.matches(&url));
```

Comment:
See https://github.com/pfernie/cookie_store/issues/16 for why `get_request_cookies()` cannot be used at the moment.

Reply:
This is now documented as of version v0.14.1. See https://github.com/pfernie/cookie_store/issues/16#issuecomment-839981352